internal/audit/audit.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package audit
+
+import (
+	"crypto/md5"
+	"fmt"
+	"github.com/juju/loggo"
+	"strings"
+)
+
+var logger = loggo.GetLogger("audit")
+
+
+type AclOperation string
+
+const (
+	SET_READ AclOperation = "set_read"
+	SET_WRITE AclOperation = "set_write"
+)
+
+func LogAcl(entity string, operation AclOperation, final_state []string) {
+	state := make([]string, len(final_state))
+	for k, v := range final_state {
+		hasher := md5.New()
```

Comment:
I'm dubious about using md5, I'm not sure it gives you much. group names don't really contain a vast amount of entropy so it would be fairly easy to dictionary attack one.